app/controllers/Application.java

```diff
@@ -165,7 +181,23 @@ public static Result search() {
           .where()
           .idEq(appId).findUnique();
       return ok(searchPage.render(null, jobDetails.render(result)));
-    } else if (flowExecId != null && !flowExecId.isEmpty()) {
+    } else if (Utils.isSet(flowExecId)) {
+      String bestFlowExecId;
+      // check for exact match
+      bestFlowExecId = bestMatchForFlowExecIdLikeValue(flowExecId);
```

Comment:
As @shankar37 pointed out, I too am a bit skeptic about the performance with partial search.

I don't think using mysql queries with ```LIKE %something``` is scalable. The queries will slow down on large databases. Running one time queries(select/update/delete) with partial search is fine but having a dedicated search filter where we could potentially have many partial search queries looks a bit scary.

Another point I'd like to add is, there is a lag between the job completion and the report actually appearing in Dr. Elephant. Because of this lag, if users query their flow immediately after the job completes, then we will unnecessarily start firing partial search queries.

My suggestion is to remove ```%something``` type of queries.

@shankar37 , do you have any other concern?